xa30_workaround/scripts/dcmdat2niix.py

```diff
@@ -91,12 +111,40 @@ def main():
         nifti_img = nib.load(nifti_img_path)
 
         # get the shape of the nifti file
+        # we want the first dimension to me the number of echos
+        # the next three dimensions should be the volume
+        # we want to ignore the number of time points
         shape = nifti_img.shape
         rshape = list(shape[::-1])
-        rshape[0] = TEs.shape[0]  # we want # of TEs instead of frames
-
-        # now search for .dat files that neighbor the exemplar dicom file
-        dat_files = list(dicom.parent.glob("*.dat"))
+        if len(rshape) <= 3:
+            # it would appear there is only one time point
+            # prepend the number of echos to the array
+            rshape.insert(0, TEs.shape[0])
+        else:
+            # it looks like the first dimension is the number of time points
+            # replace time with number of TEs
+            rshape[0] = TEs.shape[0]
+
+        # now search for .dat files
+        if args.dat_dir is None:
+            # look for .dat files that neighbor the exemplar dicom file
+            dat_files = list(dicom.parent.glob("*.dat"))
+        else:
+            # look for .dat files in the specified directory
+            # find dat files that match the Series Instance UID of the dicom
+
+            # extract Series Instance UID of this dicom file
+            # should look something like
+            # 1.3.12.2.1107.5.2.43.166158.2023072109355378899049069.0.0.0
+            # strip off the last six characters, the .0.0.0 part
+            # should then look something like
+            # 1.3.12.2.1107.5.2.43.166158.2023072109355378899049069
+            dicom_sid = pydicom.dcmread(dicom)[0x0020, 0x000e].value[:-6]
+            print(dicom_sid)
+
+            # look for .dat files in dat_dir whose name contains the sid
+            # skip hidden files starting with a .
+            dat_files = list(args.dat_dir.glob(f"[!.]*{dicom_sid}*.dat"))
```

Comment:
Is there a reason for globing by SIUID like this? Would be simpler to just have a equivalent like on line 130 but on `dat_dir`.

As is, using the `dat_dir` option would require the SIUID in the filename (maybe this is fine? I'm not sure how the dat files are currently exported).

Reply:
If for some reason SIUID is needed maybe add another arg flag that searches by "SIUID" only or a "*". Just not sure if this would break for some people with SIUID globbing.

Reply:
I think you're wondering why we don't just do:

```python
dat_files = list(args.dat_dir.glob("*.dat"))
```

1. The `dat_dir` contains `*.dat` files for multiple scans mixed together. The SIUID is needed to figure out which `*.dat` file goes with which scan.
2. The `*.dat` files are often transferred by external hard drive, often by an RA using a Macbook that litters the directory with hidden files. Ergo the `[!.]` in the glob pattern.
3. The existing behavior for users who put the `*.dat` files in the same directory as the `*.dcm` files for a scan is unchanged and does not depend on the SIUID. This should minimize breakage for existing users.

Reply:
Ah I see. That makes sense.

I would clarify this behavior in the help text: https://github.com/vanandrew/XA30_workaround/pull/3/files#diff-1b624d9722c4ae167c04620dacefc6009431676f90c25c697cff7b64e5cfcc3bR35 for `dat_dir`